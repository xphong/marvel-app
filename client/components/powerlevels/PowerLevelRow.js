import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router';

export const PowerLevelRow = props => {
  const { characterData, index } = props;

  return (
    <tr>
      <td>
        <Link to={`/powerlevels/${index}`}>
          {characterData.Name}
        </Link>
      </td>
      <td>{characterData.averagePowerLevel}</td>
      <td>
        <a href={characterData.Profile_Link} target="_blank">
            <i className="external icon" />
        </a>
      </td>
    </tr>
  );
};

PowerLevelRow.propTypes = {
  characterData: PropTypes.object.isRequired,
  index: PropTypes.number.isRequired
};

export default PowerLevelRow;
