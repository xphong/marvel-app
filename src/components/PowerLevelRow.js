import React, { Component, PropTypes } from 'react';

export default class PowerLevelRow extends Component {
  constructor(props){
    super(props);
  }

  render() {
    const {characterData} = this.props;

    return (
      <tr>
        <td>{characterData.Name}</td>
        <td>{characterData.AveragePowerLevel}</td>
        <td>
          <a href={characterData.Profile_Link} target='_blank'>
              <i className='external icon'></i>
          </a>
        </td>
      </tr>
    );
  }
}

PowerLevelRow.propTypes = {
  characterData: PropTypes.object.isRequired
};
