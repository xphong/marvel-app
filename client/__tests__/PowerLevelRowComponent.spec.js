import React from 'react';

import PowerLevelRow from '../components/powerlevels/PowerLevelRow';

describe('PowerLevelRow Component', () => {
  it('should render PowerLevelRow', () => {
    const props = {
      characterData: {
        Name: 'Storm',
        AveragePowerLevel: 9000
      },
      index: 1
    };

    const wrapper = shallow(<PowerLevelRow {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
