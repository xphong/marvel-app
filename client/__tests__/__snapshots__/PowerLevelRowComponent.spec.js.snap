// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowerLevelRow Component should render PowerLevelRow 1`] = `
<tr>
  <td>
    <Link
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/powerlevels/1"
    >
      Storm
    </Link>
  </td>
  <td>
    9000
  </td>
  <td>
    <a
      target="_blank"
    >
      <i
        className="external icon"
      />
    </a>
  </td>
</tr>
`;
